inst/include/child_model_simulation.hpp

```diff
@@ -1409,66 +1400,66 @@ void calc_art_ltfu(int t,
                    IntermediateData<ModelVariant, real_type> &intermediate) {
   static_assert(ModelVariant::run_child_model,
                 "calc_art_ltfu can only be called for model variants where run_child_model is true");
-  constexpr auto ss = StateSpace<ModelVariant>().base;
-  constexpr auto hc_ss = StateSpace<ModelVariant>().children;
-  const auto cpars = pars.children.children;
-
-  for (int s = 0; s <ss.NS; ++s) {
-    for (int a = 0; a < pars.base.options.p_idx_fertility_first; ++a) {
-      for (int hd = 0; hd < hc_ss.hc1DS; ++hd) {
-        for (int cat = 0; cat < hc_ss.hcTT; ++cat) {
-
-        if (a < hc_ss.hc2_agestart) {
-          intermediate.children.hc_hiv_total(hd, a, s) += state_next.children.hc1_hiv_pop(hd, cat, a, s);
-        } else if (hd < (hc_ss.hc2DS)) {
-          intermediate.children.hc_hiv_total(hd, a, s) += state_next.children.hc2_hiv_pop(hd, cat, a - hc_ss.hc2_agestart, s);
-        }
+  constexpr auto ss_b = StateSpace<ModelVariant>().base;
+  constexpr auto ss_c = StateSpace<ModelVariant>().children;
+  const auto& p_hc = pars.children.children;
+  const auto& p_op = pars.base.options;
+  auto& n_hc = state_next.children;
+  auto& i_hc = intermediate.children;
 
-        } // end hc_ss.hcTT
-      } //end hc_ss.hc1DS
+  for (int s = 0; s < ss_b.NS; ++s) {
+    for (int a = 0; a < p_op.p_idx_fertility_first; ++a) {
+      for (int hd = 0; hd < ss_c.hc1DS; ++hd) {
+        for (int cat = 0; cat < ss_c.hcTT; ++cat) {
+          if (a < ss_c.hc2_agestart) {
+            i_hc.hc_hiv_total(hd, a, s) += n_hc.hc1_hiv_pop(hd, cat, a, s);
+          } else if (hd < ss_c.hc2DS) {
+            i_hc.hc_hiv_total(hd, a, s) += n_hc.hc2_hiv_pop(hd, cat, a - ss_c.hc2_agestart, s);
+          }
```

Comment:
we gotta do something about this pattern, ive been thinking a helper function that returns the correct `hiv_pop` tensor based on `a` and `hd` and then we just use it as normal, in that function we also overload the indexing operator so that it subtracts `ss_c.hc2_agestart` from `a` automatically?